pkg/routes/logs_test.go

```diff
@@ -0,0 +1,57 @@
+package routes
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+)
+
+func TestPreCheckLogFileNameLength(t *testing.T) {
+	oversizeFileName := fmt.Sprintf("%0256s", "a")
+	correctFileName := fmt.Sprintf("%0255s", "a")
+
+	// check oversize filename
+	isOversize := preCheckLogFileNameLength(oversizeFileName)
+	if !isOversize {
+		t.Error("failed to check oversize filename")
+	}
+
+	// check correct filename when file is not exist
+	NotOversize := preCheckLogFileNameLength(correctFileName)
+	if NotOversize {
+		t.Error("failed to check allowable length filename")
+	}
+
+	// create test file with correct name and check it
+	_, err := os.Create(correctFileName)
+	if err != nil {
+		t.Error("failed to create test file")
+	}
+	defer os.Remove(correctFileName)
+	NotOversize = preCheckLogFileNameLength(correctFileName)
+	if NotOversize {
+		t.Error("failed to check allowable length filename")
+	}
+}
+
+func TestHttpServeFile(t *testing.T) {
+	logFilePath := "/var/log"
+	oversizeFileName := fmt.Sprintf("%s/%0256s", logFilePath, "a")
+	correctFileName := fmt.Sprintf("%s/%0255s", logFilePath, "a")
+
+	request, _ := http.NewRequest("", "", nil)
+
+	w1 := httptest.NewRecorder()
+	http.ServeFile(w1, request, oversizeFileName)
+	if w1.Code != http.StatusInternalServerError {
```

Comment:
for oversizeFileName, the response should be 404 instead of 500, no?

Reply:
Actually, inside http.ServeFile, expect NotExist and Forbidden, the default error code is 500, and 500 is not the code we want here, preCheckLogFileNameLength is here to solve this particular situation. 
![image](https://user-images.githubusercontent.com/39937150/119071155-e263b280-ba1b-11eb-85c8-41e369adf3f5.png)
And I want to tell the reason why need preCheckLogFileNameLength by this testcase.
add some comments now, make sense?

Reply:
Or should I just delete those test about http.serveFile, in case they cause misunderstanding?

Reply:
How about "422 Unprocessable Entity"? 404 seems a bit misleading here.

Reply:
I would focus on tests that show the desired behavior and show that the fix results in that desired behavior. I found this test confusing.

Reply:
> How about "422 Unprocessable Entity"? 404 seems a bit misleading here.

sounds good!

Reply:
> I would focus on tests that show the desired behavior and show that the fix results in that desired behavior. I found this test confusing.

You are absolutely right, http.server in test file is confusing. Now I remove it and add some comments in func logFileNameIsTooLong() to explain the check, should be easier to read now. What do you think?

Reply:
> > How about "422 Unprocessable Entity"? 404 seems a bit misleading here.
> 
> sounds good!

Oh wait, I'm sorry. It looks like this check on a file path that is not yet matched up with an existing file.  So if the stat fails, the file cannot possibly exist?  You might have been right to use 404 to begin with. My apologies. LGTM otherwise

Reply:
yeah, 404 maybe more suitable here.
Change it back now, thanks for reviewing!